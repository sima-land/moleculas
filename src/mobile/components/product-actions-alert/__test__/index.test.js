import React from 'react';
import { mount } from 'enzyme';
import { ProductActionsAlert } from '..';
import { noop } from 'lodash';
import StatementSVG from '@sima-land/ui-quarks/icons/24x24/Stroked/statement';

describe('ProductActionsContent', () => {
  it('renders properly', () => {
    const wrapper = mount(
      <ProductActionsAlert
        buttons={[
          { text: 'text', href: 'href', onClick: noop, icon: <StatementSVG /> },
          { text: 'text', href: 'href' },
          { text: 'text' },
        ]}
        alertProps={{
          title: 'title',
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('handlers should works properly', () => {
    const onClose = jest.fn();
    const wrapper = mount(
      <ProductActionsAlert
        buttons={[
          { text: 'text', href: 'href', onClick: noop, icon: <StatementSVG /> },
          { text: 'text', href: 'href' },
          { text: 'text' },
        ]}
        onClose={onClose}
      />
    );
    expect(onClose).not.toHaveBeenCalled();
    wrapper.find('CleanButton').at(0).simulate('click');
    expect(onClose).toHaveBeenCalled();
  });
});
