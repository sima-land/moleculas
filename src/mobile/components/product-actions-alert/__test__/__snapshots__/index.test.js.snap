// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductActionsContent renders properly 1`] = `
<ProductActionsAlert
  alertProps={
    Object {
      "title": "title",
    }
  }
  buttons={
    Array [
      Object {
        "href": "href",
        "icon": <ForwardRef(StatementSVG) />,
        "onClick": [Function],
        "text": "text",
      },
      Object {
        "href": "href",
        "text": "text",
      },
      Object {
        "text": "text",
      },
    ]
  }
>
  <Alert
    footer={
      <CleanGroup>
        <CleanButton>
          Закрыть
        </CleanButton>
      </CleanGroup>
    }
    title="title"
  >
    <Portal>
      <Portal
        containerInfo={
          <div>
            <div
              class="overlay"
              data-testid="alert:overlay"
              style="z-index: 100;"
            >
              <div
                class="alert"
                data-testid="alert"
              >
                <div
                  class="nav-bar header"
                  style="height: 56px;"
                >
                  <div
                    class="main"
                  >
                    <div
                      class="title truncate"
                    >
                      title
                    </div>
                  </div>
                </div>
                <div
                  class="main"
                >
                  <div
                    class="buttons"
                  >
                    <a
                      class="link button gray87"
                      data-testid="anchor"
                      href="href"
                    >
                      <div
                        class="icon-wrapper"
                      >
                        <svg
                          height="24"
                          viewBox="0 0 24 24"
                          width="24"
                        >
                          <path
                            clip-rule="evenodd"
                            d="M3 5a3 3 0 013-3h12a3 3 0 013 3v14a3 3 0 01-3 3H6a3 3 0 01-3-3V5zm3-1a1 1 0 00-1 1v14a1 1 0 001 1h12a1 1 0 001-1V5a1 1 0 00-1-1H6z"
                            fill-rule="evenodd"
                          />
                          <path
                            clip-rule="evenodd"
                            d="M7 16a1 1 0 011-1h5a1 1 0 110 2H8a1 1 0 01-1-1zM7 12a1 1 0 011-1h7a1 1 0 110 2H8a1 1 0 01-1-1zM7 8a1 1 0 011-1h7a1 1 0 110 2H8a1 1 0 01-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </div>
                      text
                    </a>
                    <a
                      class="link button gray87"
                      data-testid="anchor"
                      href="href"
                    >
                      <div
                        class="icon-wrapper"
                      />
                      text
                    </a>
                    <a
                      class="link button gray87"
                      data-testid="anchor"
                    >
                      <div
                        class="icon-wrapper"
                      />
                      text
                    </a>
                  </div>
                </div>
                <div
                  class="inner-border-top"
                >
                  <div
                    class="clean-group"
                  >
                    <a
                      class="link clean-button brand-blue"
                      data-testid="anchor"
                    >
                      Закрыть
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        }
      >
        <AlertInner
          footer={
            <CleanGroup>
              <CleanButton>
                Закрыть
              </CleanButton>
            </CleanGroup>
          }
          title="title"
        >
          <div
            className="overlay"
            data-testid="alert:overlay"
            onClick={[Function]}
            style={
              Object {
                "zIndex": 100,
              }
            }
          >
            <div
              className="alert"
              data-testid="alert"
            >
              <NavBar
                className="header"
                title="title"
              >
                <div
                  className="nav-bar header"
                  style={
                    Object {
                      "height": "56px",
                    }
                  }
                >
                  <div
                    className="main"
                  >
                    <div
                      className="title truncate"
                    >
                      title
                    </div>
                  </div>
                </div>
              </NavBar>
              <div
                className="main"
              >
                <div
                  className="buttons"
                >
                  <ForwardRef(Link)
                    className="button"
                    color="gray87"
                    href="href"
                    key="0"
                    onClick={[Function]}
                  >
                    <a
                      className="link button gray87"
                      data-testid="anchor"
                      href="href"
                      onClick={[Function]}
                    >
                      <div
                        className="icon-wrapper"
                      >
                        <ForwardRef(StatementSVG)>
                          <svg
                            height={24}
                            viewBox="0 0 24 24"
                            width={24}
                          >
                            <path
                              clipRule="evenodd"
                              d="M3 5a3 3 0 013-3h12a3 3 0 013 3v14a3 3 0 01-3 3H6a3 3 0 01-3-3V5zm3-1a1 1 0 00-1 1v14a1 1 0 001 1h12a1 1 0 001-1V5a1 1 0 00-1-1H6z"
                              fillRule="evenodd"
                            />
                            <path
                              clipRule="evenodd"
                              d="M7 16a1 1 0 011-1h5a1 1 0 110 2H8a1 1 0 01-1-1zM7 12a1 1 0 011-1h7a1 1 0 110 2H8a1 1 0 01-1-1zM7 8a1 1 0 011-1h7a1 1 0 110 2H8a1 1 0 01-1-1z"
                              fillRule="evenodd"
                            />
                          </svg>
                        </ForwardRef(StatementSVG)>
                      </div>
                      text
                    </a>
                  </ForwardRef(Link)>
                  <ForwardRef(Link)
                    className="button"
                    color="gray87"
                    href="href"
                    key="1"
                  >
                    <a
                      className="link button gray87"
                      data-testid="anchor"
                      href="href"
                    >
                      <div
                        className="icon-wrapper"
                      />
                      text
                    </a>
                  </ForwardRef(Link)>
                  <ForwardRef(Link)
                    className="button"
                    color="gray87"
                    key="2"
                  >
                    <a
                      className="link button gray87"
                      data-testid="anchor"
                    >
                      <div
                        className="icon-wrapper"
                      />
                      text
                    </a>
                  </ForwardRef(Link)>
                </div>
              </div>
              <div
                className="inner-border-top"
              >
                <CleanGroup>
                  <div
                    className="clean-group"
                  >
                    <CleanButton
                      key=".0"
                    >
                      <ForwardRef(Link)
                        className="clean-button"
                        pseudo={false}
                      >
                        <a
                          className="link clean-button brand-blue"
                          data-testid="anchor"
                        >
                          Закрыть
                        </a>
                      </ForwardRef(Link)>
                    </CleanButton>
                  </div>
                </CleanGroup>
              </div>
            </div>
          </div>
        </AlertInner>
      </Portal>
    </Portal>
  </Alert>
</ProductActionsAlert>
`;
